src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -222,14 +220,12 @@ public boolean equals(Object other) {
                 return false;
             }
 
-            // state check
-            EditPersonDescriptor e = (EditPersonDescriptor) other;
-
-            return getName().equals(e.getName())
-                    && getPhone().equals(e.getPhone())
-                    && getEmail().equals(e.getEmail())
-                    && getAddress().equals(e.getAddress())
-                    && getTags().equals(e.getTags());
+            EditPersonDescriptor otherEditPersonDescriptor = (EditPersonDescriptor) other;
+            return Objects.equals(name, otherEditPersonDescriptor.name)
```

Comment:
From my understanding, Objects.equals() considers 2 nulls to be equal. Although, in this case, the name field being null would have been caught already, perhaps it would still be better to use Object.equals() or the default equals method of the respective objects?

eg.
`name.equals(otherEditPersonDescriptor.name)`

Reply:
I don't think this will occur due to requireNonNull(editPersonDescriptor);.

Reply:
it's possible for the local name field to be null. User does not need to edit all the fields of an entry so it's possible to have name being null. In this case `name.equals(...)` will throw nullpointer exception. 

So Objects.equals is used here